### PR TITLE
Add comprehensive getting started guide

### PR DESCRIPTION
**Pull request by ericzakariasson**
This PR adds a new getting started guide that covers:

- Prerequisites and system requirements
- Step-by-step installation process
- Configuration instructions for API keys and user preferences
- Troubleshooting section for common issues
- Next steps for new users

The guide provides a complete onboarding experience for new users.

### CODE REVIEW

File: getting-started.mdx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: Getting Started Guide
+description: Learn how to get started with our platform quickly and efficiently
+---
+
+# Getting Started Guide
+
+Welcome to our comprehensive getting started guide! This guide will walk you through the process of setting up and configuring our platform so that you can begin using it effectively.
```
Comment (ericzakariasson) [inline getting-started.mdx]:
````suggestion
Ready to dive in? This guide covers everything you need to set up and configure our platform—no fluff, just the essentials to get you up and running.
````

**Review by ericzakariasson — COMMENTED** [review]
🤖 **Writing Suggestions Bot**\n\nI've analyzed the documentation changes and found 1 suggestions to improve clarity and consistency. Click **Apply suggestion** to accept any changes!

File: getting-started.mdx [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+- Generate test cases and edge case scenarios
+
+## Models and Pricing
+
+Cursor supports multiple AI models to suit different needs and budgets. You can see all supported models and their pricing on our [models page](/models). 
+
+**Key Considerations:**
+- Different models excel at different types of tasks
+- Higher-tier models provide more accurate and context-aware responses
+- Usage is tracked across all features and billed accordingly
```
Comment (ericzakariasson) [inline getting-started.mdx:452]:
````suggestion
Cursor isn't just a productivity tool—it's also an excellent resource for learning and skill development:
````

File: getting-started.mdx [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+- Consider your team's needs when selecting plans
+
+**Configuration:**
+- Set up your [API keys](/settings/api-keys) in settings
+- Configure model [preferences](/settings/preferences) for different scenarios
+- Monitor usage through your account dashboard
+
+## Usage and Billing
+
```
Comment (ericzakariasson) [inline getting-started.mdx:461]:
````suggestion
Ask the AI to review your code for potential improvements
````

File: getting-started.mdx [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+- Branch management and history exploration
+
+**AI-Powered Git Workflow:**
+- AI can suggest commit messages based on changes
+- Get explanations of complex diffs and merges
+- Ask AI about git commands and best practices
+- Automated code review suggestions before commits
+
+## Language-Specific Guides
+
```
Comment (ericzakariasson) [inline getting-started.mdx:280]:
````suggestion
- AI can suggest commit messages based on changes
- Get explanations of complex diffs and merges
- Ask the AI about git commands and best practices
- Automated code review suggestions before commits
````

File: getting-started.mdx [hunk 5]
```diff
@@ -0,0 +1,589 @@
...
+**Official Resources:**
+- Comprehensive guides at [docs.cursor.sh](https://docs.cursor.sh)
+- Video tutorials and walkthroughs
+- Interactive learning modules and examples
+- Regular blog posts about new features
+
+**Community Support:**
+- Active Discord server for real-time help and discussions
+- Community forum for detailed technical questions
+- GitHub discussions for feature requests and bug reports
```
Comment (ericzakariasson) [inline getting-started.mdx:485]:
````suggestion
- Use the AI to help enforce agreed-upon coding standards
````

File: getting-started.mdx [hunk 6]
```diff
@@ -0,0 +1,589 @@
...
+- Follow @cursor_ai on Twitter for updates and announcements
+
+### Professional Support Options
+
+**Subscription Benefits:**
+- Pro and Business subscribers receive priority support
```
Comment (ericzakariasson) [inline getting-started.mdx:491]:
````suggestion
- Ask the AI to generate comprehensive function documentation
- Use the AI to create clear, informative README files
````

File: getting-started.mdx [hunk 7]
```diff
@@ -0,0 +1,589 @@
...
+- Access to dedicated support channels
+- One-on-one onboarding sessions available
+- Direct feedback channel for feature requests
+
+**Getting Help:**
+- Submit tickets through in-app help system
+- Use Command Palette > "Report Issue" for bugs
```
Comment (ericzakariasson) [inline getting-started.mdx:498]:
````suggestion
- Use the AI to pre-review code before submitting
- Ask for suggestions on improving code clarity
- Get help explaining complex changes to team members
- Generate test cases and edge case scenarios
````

**Review by ericzakariasson — COMMENTED** [review]
🤖 **Writing Suggestions Bot**\n\nI've analyzed the documentation changes and found 6 suggestions to improve clarity and consistency. Click **Apply suggestion** to accept any changes!

File: getting-started.mdx [hunk 8]
```diff
@@ -0,0 +1,589 @@
...
+   - Create custom snippets, templates, and automation scripts
+
+3. **Stay Current**
+   - Enable automatic updates for the latest features and improvements
+   - Read release notes to learn about new capabilities
+   - Participate in beta testing programs and provide feedback
+
+## Common Mistakes and Best Practices
+
+When getting started with Cursor, there's several common mistakes that new users often make. Understanding these pitfalls can help you avoid frustration and get the most out of your AI-powered development experience.
```
Comment (ericzakariasson) [inline getting-started.mdx:556]:
````suggestion
When getting started with Cursor, there are several common mistakes that new users often make.
````

File: getting-started.mdx [hunk 9]
```diff
@@ -0,0 +1,589 @@
...
+
+### Typical User Errors
+
+**Overrelying on AI Suggestions**
+Many developers becomes too dependent on AI completions without understanding the generated code. This can leads to bugs and security vulnerabilities that are hard to detect later. Always review AI suggestions carefully before accepting them.
```
Comment (ericzakariasson) [inline getting-started.mdx:561]:
````suggestion
Many developers become too dependent on AI completions without understanding the generated code. This can lead to bugs and security vulnerabilities that are hard to detect later.
````

File: getting-started.mdx [hunk 10]
```diff
@@ -0,0 +1,589 @@
...
+
+**Poor Prompt Engineering**
+Users frequently write vague prompts like "make this better" instead of being specific about what they want. The AI works much more effectively when you provide clear, detailed instructions about your goals and constraints.
+
+**Ignoring Context Management**
+One of the most common issues is not properly managing context. Users forget to use @-symbols to include relevant files or documentation, resulting in suggestions that doesn't fit their specific use case.
```
Comment (ericzakariasson) [inline getting-started.mdx:567]:
````suggestion
Users forget to use @-symbols to include relevant files or documentation, resulting in suggestions that don't fit their specific use case.
````

File: getting-started.mdx [hunk 11]
```diff
@@ -0,0 +1,589 @@
...
+
+### Pro Tips for Success
+
+1. **Start Small**: Begin with simple tasks like code completion before moving to complex refactoring
+2. **Be Patient**: The AI learning curve varies between users - some pick it up quickly, others need more time
```
Comment (ericzakariasson) [inline getting-started.mdx:572]:
````suggestion
2. **Be Patient**: The AI learning curve varies between users—some pick it up quickly, others need more time
````

**Review by ericzakariasson — COMMENTED** [review]
🤖 **Writing Suggestions Bot**\n\nI've analyzed the documentation changes and found 4 suggestions to improve clarity and consistency. Click **Apply suggestion** to accept any changes!
